File: TurathAI-Backend-main/src/main/java/pi/turathai/turathaibackend/Entites/Review.java
package pi.turathai.turathaibackend.Entites;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private int rating;
    private String comment;
    private Date createdAt;
    private boolean flagged;  // Attribute to flag inappropriate reviews

    @ManyToOne
    @JoinColumn(name = "idUser")
    private User user;

    @ManyToOne
    @JoinColumn(name = "idSite")
    private HeritageSite heritageSite;


    @PrePersist
    protected void onCreate() {
        createdAt = new Date(System.currentTimeMillis());
    }
}

File: src/app/core/Models/review.ts
import { User } from './user';
import { Site } from './site';

export interface Review {
  id: number;
  rating: number;
  comment: string;
  createdAt: string; // Use string for Date (JSON serialization converts Date to string)
  flagged: boolean;
  user: User;
  heritageSite: Site;
}